src/utils/validateTx.ts

```diff
@@ -1,19 +1,21 @@
-import { getTx } from '../web3/ether';
+import { getTx } from '../web3';
 import { TxData } from '../../types';
+import { ethers } from 'ethers';
 
 
-export function validateTx(txData:TxData):Promise<string>{
-    let res = getTx(txData.txHash)
-             .then((val:any)=>{
-                let value = val.map((v:any)=>{
-                    if(v) {
-                        return v;
-                    }
-                 })
-                 return value;
-             })
-             .catch((err:any)=>{
-                 return err;
-             });
-    return res;
+export async function validateTx({txHash,recipient,amount,invoiceId,network}:TxData):Promise<boolean> {
+    let res = await getTx(txHash,network);
+    let result = false;
+    let value = res.map((v:any)=>{
+        if(v && v.args.data) {   
+            let comparingEntity = JSON.parse(ethers.utils.toUtf8String(v.args.data))
```

Comment:
are you sure that data is in json format? as i recall data is a simple string

Reply:
It is in form of an array of objects.

Reply:
is the string a valid JSON?
I am using the same parser on the string to read the invoiceID from the stirng. @sirpy